internal/service/feature_usage_tracking.go

```diff
@@ -676,8 +764,13 @@ func (s *featureUsageTrackingService) findMatchingPricesForEvent(
 			continue
 		}
 
-		// Check meter filters
-		if !s.checkMeterFilters(event, meter.Filters) {
```

Comment:
this is a very light weight static check so don't think caching it is going to add a lot of value